platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -199,8 +200,43 @@ private Browser(final Set<Integer> localNodesToStart) throws IOException {
                 .withConfigDataType(MetricsConfig.class)
                 .withConfigDataType(PrometheusConfig.class)
                 .withConfigDataType(OSHealthCheckConfig.class)
-                .withConfigDataType(WiringConfig.class)
-                .build();
+                .withConfigDataType(WiringConfig.class);
+
+        // The following variables are not final while being initialized.
+        ApplicationDefinition appDefinitionInit = null;
+        Map<Long, SwirldMain> appMainsInit = null;
+        Configuration config = null;
+        Exception exception = null;
+        try {
+            // Load config.txt file, parse application jar file name, main class name, address book, and parameters
+            appDefinitionInit = ApplicationDefinitionLoader.load(configurationProperties, localNodesToStart);
+
+            // Load all SwirldMain instances for locally run nodes.
+            appMainsInit = loadSwirldMains(appDefinitionInit, localNodesToStart);
+
+            // We assume all locally run instances would provide the same config to the configuration builder.
+            if (appMainsInit.size() > 0) {
+                appMainsInit.get(0L).updateConfigurationBuilder(configurationBuilder);
+            }
+
+            config = configurationBuilder.build();
+        } catch (final Exception e) {
+            logger.error(EXCEPTION.getMarker(), "Failed to load application due to exception.", e);
+            exception = e;
+        } finally {
+            // this.configuration needs to be initialized before the constructor returns or throws exception.
+            this.configuration = config;
```

Comment:
I do not get this behavior. Why not just rethrow the exception and simply set this.configuration in the last line of the try block.

Reply:
I agree... failure to load configuration should be considered a catastrophic failure that we do not attempt to recover from. A node with improper configuration is more likely than not to cause disturbances on the network.

Reply:
I have messed with the code block structurally and I keep getting compile errors because the configuration is final and needs to be assigned before exiting the constructor.   Java doesn't like the following: 
```
try {
    this.configuration = foo();
} catch (Exception e){
    this.configuration = null;
}
```
I get a compile error saying that `this.configuration` might have been assigned to in the try{} block.  

I'll see if I can reorganize some more.  Is throwing on the constructor an acceptable resolution?   (not having the try block.)

Reply:
I am now throwing a runtime exception that kills building the Browser constructor.  code is reorganized in the next commit.